pkg/client/unversioned/clientcmd/api/v1/types.go

```diff
@@ -54,7 +54,10 @@ type Preferences struct {
 type Cluster struct {
 	// Server is the address of the kubernetes cluster (https://hostname:port).
 	Server string `json:"server"`
-	// APIVersion is the preferred api version for communicating with the kubernetes cluster (v1, v2, etc).
+	// DEPRECATED: APIVersion is the preferred api version for communicating with the kubernetes cluster (v1, v2, etc).
+	//  Because a cluster can run multiple API groups and potentially multiple versions of each, it no longer makes sense to specify
+	//  a single value for the cluster version. This field isn't really needed anyway, so we're deprecating it without a replacement.
+	//  It will be ignored if it is present.
```

Comment:
> "experimental/v1alpha1" - this means no preference on the v1 api version, but if you get something from experimental, you must use the v1alpha1 version.

I like this idea in general, but I'm not sure of "must" versus "must if available".  I don't think we have to debate it before this changes in, but we currently have "must if available" and I've been pretty pleased with its operation.

I am a little concerned over the data structure choice.  If we placed this is a map, then it would be impossible to have multiple mappings for "experimental".  What happens in this scenario if you have "experimental/v1" and "experimental/v2"?